packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -162,10 +168,12 @@ class DropdownMenu<T> extends StatefulWidget {
     this.focusNode,
     this.requestFocusOnTap,
     this.expandedInsets,
+    this.filterCallback,
     this.searchCallback,
     required this.dropdownMenuEntries,
     this.inputFormatters,
-  });
+  }) :
+        assert(filterCallback == null || enableFilter);
```

Comment:
Nit: I think the assert can go on the line above.

Reply:
Nit: Maybe add an explanation to the assert: 'filterCallback can only be used when enableFilter is true'. Not sure if that's actually helpful or not, up to you.

Reply:
If you want to land another PR after this one, we should probably have a similar assert for searchCallback and enableSearch :grin:

Reply:
> Nit: Maybe add an explanation to the assert: 'filterCallback can only be used when enableFilter is true'. Not sure if that's actually helpful or not, up to you.

I wrote a comment right above the assert in a previous commit, but @QuncCccccc suggested to move the explanation in the `filterCallback` documentation